qa/mixed-cluster/build.gradle

```diff
@@ -17,58 +17,88 @@
  * under the License.
  */
 
-import org.elasticsearch.gradle.test.RestIntegTestTask
 import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.testclusters.RestTestRunnerTask
 
+apply plugin: 'elasticsearch.testclusters'
 apply plugin: 'elasticsearch.standalone-test'
 
-// This is a top level task which we will add dependencies to below.
-// It is a single task that can be used to backcompat tests against all versions.
-task bwcTest {
+tasks.register("bwcTest") {
   description = 'Runs backwards compatibility tests.'
   group = 'verification'
 }
 
-for (Version version : bwcVersions.wireCompatible) {
-  String baseName = "v${version}"
+configurations {
+  restSpec
+}
 
-  Task mixedClusterTest = tasks.create(name: "${baseName}#mixedClusterTest", type: RestIntegTestTask) {
-    mustRunAfter(precommit)
-    includePackaged = true
+dependencies {
+  restSpec project(':rest-api-spec')
+}
+
+processTestResources {
+  from ({ zipTree(configurations.restSpec.singleFile) })
+  dependsOn configurations.restSpec
+}
+
+for (Version bwcVersion : bwcVersions.wireCompatible) {
+  if (bwcVersion == Version.current()) {
+    // Not really a mixed cluster
+    continue ;
   }
 
+  String baseName = "v${bwcVersion}"
+
   /* This project runs the core REST tests against a 4 node cluster where two of
      the nodes has a different minor.  */
-  Object extension = extensions.findByName("${baseName}#mixedClusterTestCluster")
-  configure(extension) {
-    numNodes = 4
-    numBwcNodes = 2
-    bwcVersion = version
-  }
+  testClusters {
+    "${baseName}" {
+      versions = [ bwcVersion.toString(), project.version ]
+      numberOfNodes = 4
 
-  Task versionBwcTest = tasks.create(name: "${baseName}#bwcTest") {
-    dependsOn = [mixedClusterTest]
+      setting 'path.repo', "${buildDir}/cluster/shared/repo/${baseName}"
+      javaHome = project.file(project.ext.runtimeJavaHome)
+    }
   }
 
-  if (project.bwc_tests_enabled) {
-    bwcTest.dependsOn(versionBwcTest)
+  tasks.register("${baseName}#mixedClusterTest", RestTestRunnerTask) {
+    useCluster testClusters."${baseName}"
+    mustRunAfter(precommit)
+    doFirst {
```

Comment:
Do we want to keep all this logging here? I'm not against it, just want to make sure it's not leftover debugging.

Reply:
We need to get the endpoints to make sure the cluster is up before we force an upgrade.  It's not ideal but something we will address with the improvements we are considering.